### PR TITLE
Handle windows linebreaks in getSourceFileImportLocation

### PR DESCRIPTION
**Pull request by amcasey**

### CODE REVIEW

File: src/services/utilities.ts [hunk 1]
```diff
@@ -1335,18 +1335,30 @@ namespace ts {
...
+                position = range.end;
+                AdvancePastLineBreak();
                 continue;
             }
             break;
         }
         return position;
+
+        function AdvancePastLineBreak() {
+            if (text.charCodeAt(position) === 0xD) {
```
Comment (mhegazy) [inline src/services/utilities.ts]: `CharacterCodes.carriageReturn`

File: src/services/utilities.ts [hunk 2]
```diff
@@ -1335,18 +1335,30 @@ namespace ts {
...
+                position++;
+            }
+
+            if (text.charCodeAt(position) === 0xA) {
```
Comment (mhegazy) [inline src/services/utilities.ts]: `Charactercodes.lineFeed`

Comment (mhegazy) [reply src/services/utilities.ts]: use `isLineBreak` to handle other line break characters like \u2028 ans 29

File: src/services/utilities.ts (hunk 1 above)
Comment (weswigham) [inline src/services/utilities.ts]: `ts.characterCodes.carriageReturn` over 0xD?

File: src/services/utilities.ts (hunk 2 above)
Comment (weswigham) [inline src/services/utilities.ts]: Same here, I think it's more clear if the character codes enum is used.

File: src/services/utilities.ts [hunk 3]
```diff
@@ -1335,18 +1335,30 @@ namespace ts {
...
-                position = range.end + 1;
+                position = range.end;
+                AdvancePastLineBreak();
                 continue;
             }
             break;
         }
         return position;
+
+        function AdvancePastLineBreak() {
```
Comment (mhegazy) [inline src/services/utilities.ts]: would `skipTrivia` be sufficient here?

Comment (amcasey) [reply src/services/utilities.ts]: Wouldn't that skip other whitespace and comments?

File: src/services/utilities.ts (hunk 1 above)
Comment (amcasey) [reply src/services/utilities.ts]: Nifty.

File: src/services/utilities.ts (hunk 3 above)
Comment (weswigham) [reply src/services/utilities.ts]: Though I am confused as to why this is upper camel case instead of lower camel case like every other helper function .

Comment (amcasey) [reply src/services/utilities.ts]: That would be because I'm a C# dev. :wink:

Comment (mhegazy) [reply src/services/utilities.ts]: but you are at the end of the line anyways.. so comments and white spaces are not an issue here..

Comment (amcasey) [reply src/services/utilities.ts]: Good point.  I notice, though, that `skipTrivia` doesn't handle the other linebreak characters.  Unless you object, I'll leave it like this for the sake of explicitness.

**Pull request merged**